src/VisualStudio/Core/Impl/RoslynVisualStudioWorkspace.cs

```diff
@@ -124,8 +124,22 @@ internal override IInvisibleEditor OpenInvisibleEditor(IVisualStudioHostDocument
         {
             // We need to ensure the file is saved, only if a global undo transaction is open
             var globalUndoService = this.Services.GetService<IGlobalUndoService>();
-            bool needsSave = globalUndoService.IsGlobalTransactionOpen(this);
-            bool needsUndoDisabled = needsSave && this.Services.GetService<IGeneratedCodeRecognitionService>().IsGeneratedCode(this.CurrentSolution.GetDocument(hostDocument.Id));
+            var needsSave = globalUndoService.IsGlobalTransactionOpen(this);
+
+            var needsUndoDisabled = false;
+            if (needsSave)
+            {
+                if (this.CurrentSolution.ContainsDocument(hostDocument.Id))
```

Comment:
There's a comment in the else about one case, which is AdditionalDocuments. @mavasani mentioned something in the bug about being resilient against documents that had been removed from the solution, but I don't know exactly how that happens.